rasa/core/featurizers/precomputation.py

```diff
@@ -0,0 +1,414 @@
+        """Adds all relevant messages that can be derived from the given events.
+
+        That is, each action name, action text, user text and intent that can be
+        found in the given events will be turned into a (separate) message and added
+        to this container.
+
+        Args:
+          events: list of events to extract the substate from
+        """
+        for event in events:
+            key_value_list = []
+            if isinstance(event, UserUttered):
+                key_value_list = [(TEXT, event.text), (INTENT, event.intent_name)]
+            elif isinstance(event, ActionExecuted):
+                key_value_list = [
+                    (ACTION_TEXT, event.action_text),
+                    (ACTION_NAME, event.action_name),
+                ]
+            for key, value in key_value_list:
+                if value is not None:
+                    self.add(Message(data={key: value}))
+
+
+class CoreFeaturizationInputConverter(GraphComponent):
+    """Provides data for the featurization pipeline.
+
+    During training as well as during inference, the converter de-duplicates the given
+    data (i.e. story graph or list of messages) such that each text and intent from a
+    user message and each action name and action text appears exactly once.
+    """
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> CoreFeaturizationInputConverter:
+        """Creates a new instance (see parent class for full docstring)."""
+        return cls()
+
+    def convert_for_training(
+        self, domain: Domain, story_graph: StoryGraph,
+    ) -> TrainingData:
+        """Creates de-duplicated training data.
+
+        Each possible user text and intent and each action name and action text
+        that can be found in the given domain and story graph appears exactly once
+        in the resulting training data. Moreover, each item is contained in a separate
+        messsage.
+
+        Args:
+           domain: the domain
+           story_graph: a story graph
+        Returns:
+           training data
+        """
+        container = MessageContainerForCoreFeaturization()
+
+        # collect all action and user (intent-only) substates known from domain
+        container.derive_messages_from_domain_and_add(domain=domain)
+
+        # collect all substates we see in the given data
+        all_events = (
+            event
+            for step in story_graph.story_steps
+            for event in step.events
+            if isinstance(event, UserUttered)
+            # because all action names and texts are known to the domain
+        )
+        container.derive_messages_from_events_and_add(events=all_events)
+
+        # Reminder: in case of complex recipes that train CountVectorizers, we'll have
```

Comment:
True -  Shall I remove that reminder, or shall we keep it until that is fixed there?

Reply:
Do you know if there is an issue open for it yet? If not we should create one.

Reply:
I think @aeshky  is working on migrating the count vectorizer - we just fix that there (i.e. let the count vectorizer not break because it hasn't been trained but instead just not add any features to the message, like in other components), no?

Reply:
Yes - sounds good.